source/keymanweb.js

```diff
@@ -3542,16 +3574,97 @@
+       * 
+       * We place it here so that it loads after most of the other UI loads, reducing the MutationObserver's overhead.
+       */
+
+      var observationTarget = document.querySelector('body');
+      keymanweb.mutationObserver = new MutationObserver(function(mutations) 
+        {
+          var dirtyFlag = false; // Notes if we need to recompute our .sortedInputs array.
+
+          for(var i=0; i < mutations.length; i++)
+          {
+            mutation = mutations[i];
+            
+            for(var j = 0; j < mutation.addedNodes.length; j++)
+            {
+              var addedNode = mutation.addedNodes[j];
+              var lcTagName = "";
+              
+              if(addedNode.tagName) {
+                lcTagName = addedNode.tagName.toLowerCase();
+              }
+              
+              var childAdditions = [];
+
+              // Will need to handle this in case of child elements in a newly-added element with child elements.
+              if(addedNode.getElementsByTagName) 
+              {
+                childAdditions = childAdditions.concat(
+                  addedNode.getElementsByTagName('input'),
+                  addedNode.getElementsByTagName('textarea'),
+                  addedNode.getElementsByTagName('iframe'),
+                );
+              }
+
+              if(lcTagName == 'input' || lcTagName == 'textarea' || lcTagName == 'iframe')
+              {
+                dirtyFlag = true;
+                keymanweb._MutationAdditionObserved(addedNode);
+              }
+
+              for(k = 0; k < childAdditions.length; k++)
+              {
+                dirtyFlag = true;
+                keymanweb._MutationAdditionObserved(childAdditions[k]);
+              }
+            }
+
+            // There also exists a 'mutation.removedNodes' array.  We'll need to address that for issue #63.
+
+            // After all mutations have been handled, we need to recompile our .sortedInputs array.
+            if(dirtyFlag) {
+              keymanweb.listInputs();
+            }
+          }
+        });
+
+      var observationConfig = { childList: true, subtree: true };
+      keymanweb.mutationObserver.observe(observationTarget, observationConfig);
+
     // Set exposed initialization flag to 2 to indicate deferred initialization also complete
     keymanweb['initialized']=2;
-  }  
+  }
+
+  // Used by the MutationObserver event handler to properly setup any elements dynamically added to the document post-initialization.
+  keymanweb._MutationAdditionObserved = function(Pelem)
+  {
+    if(!device.touchable)
+    {
+      // keymanweb.attachToControl is written to handle iframes, but setupDesktopElement is not.
+      if(keymanweb.setupDesktopElement(Pelem)) {
+        keymanweb.attachToControl(Pelem);
+      } else if(Pelem.tagName.toLowerCase() == 'iframe') {
+        //Problem:  the iframe is loaded asynchronously, and we must wait for it to load fully before hooking in.
```

Comment:
[It may not be quite that simple.](https://stackoverflow.com/questions/3698200/window-onload-vs-document-ready)  **tl;dr**:  'ready' comes before 'onload'.

If we call `attachToControl` when we're 'ready', but haven't attached, there won't be any hooks for keyboarding.  That being said, once onload triggers, if we haven't disabled that internal hook... well, it should be fine to run `attachToControl` again and successfully gain hooks if it's idempotent, right?

Of course, there's making sure that `attachToControl` _is_ idempotent.